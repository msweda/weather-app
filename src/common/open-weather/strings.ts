export const noGeolocation = 'Geolocation must be enabled to use this feature.'
export const weatherFailed =
  'Weather information is not available at this time.'
export const forecastFailed =
  'Forecast information is not available at this time.'
export const weatherIconUrl = (weatherIcon: string) =>
  `http://openweathermap.org/img/w/${weatherIcon}.png`
