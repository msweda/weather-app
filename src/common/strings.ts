export const appHeading = 'Matthew Sweda\'s Weather App';
export const phoneNumber = '(248)-770-7487';
export const email = 'msweda.developer@gmail.com';
export const location = 'Detroit, MI';
export const copyright = (timestamp: number) => `Â© Matthew Sweda ${new Date(timestamp).getFullYear()}`;
export const emailLink = `mailto:${email}`;
export const linkedInLink = 'https://www.linkedin.com/in/matthew-sweda-118570ba/';
export const instagramLink = 'https://www.instagram.com/mattsweda/';