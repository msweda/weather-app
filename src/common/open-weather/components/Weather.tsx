import { createStyles, WithStyles, withStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import * as React from 'react'
import { Weather as WeatherInterface } from '../interfaces'
import { weatherIconUrl } from '../strings'

class Weather extends React.Component<Props> {
  render() {
    const { classes, weather } = this.props
    return (
      <div className={classes.weatherIcons}>
        {weather.map((element, index) => (
          <div
            key={`weather-array-element-${index}`}
            className={classes.weatherArrayElement}
          >
            <img src={weatherIconUrl(element.icon)} width="30" height="30" />
            <Typography color="primary">{element.main}</Typography>
          </div>
        ))}
      </div>
    )
  }
}

const styles = () =>
  createStyles({
    weatherIcons: {
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
    },
    weatherArrayElement: {
      margin: 10,
    },
  })

interface Props extends WithStyles<typeof styles> {
  weather: WeatherInterface
}

export default withStyles(styles)(Weather)
