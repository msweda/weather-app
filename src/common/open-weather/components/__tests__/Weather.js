import { createShallow } from '@material-ui/core/test-utils'
import Typography from '@material-ui/core/Typography'
import * as React from 'react'
import { weatherIconUrl } from '../../strings'
import Weather from '../Weather'

describe('<Weather />', () => {
  let shallow
  let wrapper

  beforeEach(() => {
    shallow = createShallow({ untilSelector: Weather })
    wrapper = shallow(<Weather weather={mockWeather} />)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render an icon and text for each weather array element', () => {
    const weather = wrapper.instance().props.weather
    for (let i = 0; i < weather.length; i++) {
      const img = wrapper.find('img').at(i)
      expect(img.exists()).toBeTruthy()
      expect(img.props().src).toEqual(weatherIconUrl(weather[i].icon))
      const typography = wrapper.find(Typography).at(i)
      expect(typography.exists()).toBeTruthy()
      expect(typography.children().text()).toEqual(weather[i].main)
    }
  })
})

const mockWeather = [
  {
    id: 521,
    main: 'Rain',
    description: 'shower rain',
    icon: '09n',
  },
  {
    id: 701,
    main: 'Mist',
    description: 'mist',
    icon: '50n',
  },
]
