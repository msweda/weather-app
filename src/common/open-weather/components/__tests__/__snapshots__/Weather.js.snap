// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Weather /> should match snapshot 1`] = `
<div
  className="Weather-weatherIcons-1"
>
  <div
    className="Weather-weatherArrayElement-2"
    key="weather-array-element-0"
  >
    <img
      height="30"
      src="http://openweathermap.org/img/w/09n.png"
      width="30"
    />
    <WithStyles(Typography)
      color="primary"
    >
      Rain
    </WithStyles(Typography)>
  </div>
  <div
    className="Weather-weatherArrayElement-2"
    key="weather-array-element-1"
  >
    <img
      height="30"
      src="http://openweathermap.org/img/w/50n.png"
      width="30"
    />
    <WithStyles(Typography)
      color="primary"
    >
      Mist
    </WithStyles(Typography)>
  </div>
</div>
`;
